Add docs for CSS Scroll-driven animations

chrisdavidmills opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description

The [Scroll-driven animations spec](https://drafts.csswg.org/scroll-animations-1/) is mostly supported in Chromium 115. This PR adds content for the new properties, functions, and other syntax defined in this spec, plus some related features from [CSS Animations Level 2](https://drafts.csswg.org/css-animations-2/).

See my [research document](https://docs.google.com/document/d/1GC7uqhFl2CIyXhLXvb3ODvwnd_sEx9BPNLk7WrIuJFI/edit#) for more details as to what has been added. Note that this document includes some specific comments about items that still don't seem to be supported correctly in Chrome. In some cases, they are waiting for bugs to be fixed, but if those bugs don't get fixed by the time of launch, we might have to revert a bit of BCD data.

<!-- ✍️ Summarize your changes in one or two sentences -->

### Motivation

<!-- ❓ Why are you making these changes and how do they help readers? -->

### Additional details

<!-- 🔗 Link to release notes, vendor docs, bug trackers, source control, or other places providing more context -->

### Related issues and pull requests

- Related BCD additions: https://github.com/mdn/browser-compat-data/pull/19942/
- Related CSS syntax data additions: https://github.com/mdn/data/pull/669

Fixes https://github.com/mdn/content/issues/26544

<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request should be merged first, use "**Depends on:** #123" -->


<!-- 👷‍♀️ After submitting, go to the "Checks" tab of your PR for the build status -->

chrisdavidmills committed:
Add docs for CSS Scroll-driven animations

chrisdavidmills committed:
Merge branch 'main' into scroll-driven-animations

chrisdavidmills committed:
Deleting unwanted article

chrisdavidmills committed:
prettier fixes

chrisdavidmills committed:
Fix flaws

estelle commented on files/en-us/web/css/@keyframes/index.md:
```diff
@@ -33,6 +33,8 @@ The **`@keyframes`** CSS [at-rule](/en-US/docs/Web/CSS/At-rule) controls the int
 - {{cssxref("&lt;percentage&gt;")}}
   - : A percentage of the time through the animation sequence at which the specified keyframe should occur.
+- `<timeline-range-name>` {{cssxref("&lt;percentage&gt;")}} {{experimental_inline}}
+  - : A percentage of the time through a specific named timeline range inside the overall timeline of the animation sequence, at which the specified keyframe should occur. See the {{cssxref("animation-range")}} property page for a detailed description of the available named timeline ranges, and [CSS Scroll-driven Animations](/en-US/docs/Web/CSS/CSS_Scroll-driven_Animations) for more information o ...
```
```suggestion
  - : A percentage of the time through a specific named timeline range inside the overall timeline of the animation sequence, at which the specified keyframe should occur. See the {{cssxref("animation-range")}} property page for a detailed description of the available named timeline ranges, and [CSS scroll-driven animations](/en-US/docs/Web/CSS/CSS_scroll-driven_animations) for more information on the kinds of animations that use them.
```

MDN has moved to lower case links (title case if it starts a sentence or is on its own line)

chrisdavidmills replied:
Thanks @estelle . I've just corrected the article as instructed.

estelle commented on files/en-us/web/css/css_scroll-driven_animations/index.md:
```diff
@@ -0,0 +1,223 @@
+---
+title: CSS Scroll-driven Animations
+slug: Web/CSS/CSS_Scroll-driven_Animations
```
```suggestion
title: CSS scroll-driven animations
slug: Web/CSS/CSS_scroll-driven_animations
```

chrisdavidmills committed:
Merge branch 'main' into scroll-driven-animations

chrisdavidmills committed:
Update files/en-us/web/css/css_scroll-driven_animations/index.md

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

chrisdavidmills committed:
Correct casing of scroll-driven animations article title and links

estelle commented on files/en-us/web/css/css_scroll-driven_animations/index.md:
```diff
@@ -0,0 +1,223 @@
+
+### Scroll progress timeline-driven animation
+
+The first example shows a scroll progress timeline-driven animation — scroll the container up and down to progress the animation forwards and backwards. The position of the scrollbar directly controls the animation.
```
This content is good for a guide. The "in action" section is for examples that are likely beyond the scope of a guide example, that demonstrate multiple if not all the features defined in the specification, with little to no explanation other than "this is cool" and "toggle this feature on and off". 

Completed examples in the new module page format include animations, filter-effects, backgrounds and borders, multicol layout, and blending and composting.

chrisdavidmills replied:
OK, got it. There is a really good demo site available that Bramus put together, so I've replaced the examples with a link to that.

chrisdavidmills committed:
Update landing page to remove unnecessary demos and replace with link

chrisdavidmills committed:
Merge branch 'main' into scroll-driven-animations

dipikabh reviewed (changes requested):
Hi Chris, thanks for adding these updates! I started looking at this since I have been reviewing Estelle's work on other module landing pages. Leaving a partial review for now. I'll come back to look at the other files.

chrisdavidmills commented on files/en-us/web/css/@keyframes/index.md:
```diff
@@ -33,6 +33,8 @@ The **`@keyframes`** CSS [at-rule](/en-US/docs/Web/CSS/At-rule) controls the int
 - {{cssxref("&lt;percentage&gt;")}}
   - : A percentage of the time through the animation sequence at which the specified keyframe should occur.
+- `<timeline-range-name>` {{cssxref("&lt;percentage&gt;")}} {{experimental_inline}}
+  - : A percentage of the time through a specific named timeline range inside the overall timeline of the animation sequence, at which the specified keyframe should occur. See the {{cssxref("animation-range")}} property page for a detailed description of the available named timeline ranges, and [CSS scroll-driven animations](/en-US/docs/Web/CSS/CSS_scroll-driven_animations) for more information o ...
```
Ha, yes, this is much better, thanks! I got tripped up on a few of these descriptions, as they are quite complex.

chrisdavidmills committed:
Update files/en-us/web/css/@keyframes/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills commented on files/en-us/web/css/@keyframes/index.md:
```diff
@@ -143,7 +145,7 @@ Declarations in a keyframe qualified with `!important` are ignored.
 ### CSS animation examples
 
-See [Using CSS animations](/en-US/docs/Web/CSS/CSS_Animations/Using_CSS_animations) for examples.
+See [Using CSS animations](/en-US/docs/Web/CSS/CSS_animations/Using_CSS_animations) and [Animate elements on scroll with Scroll-driven animations](https://developer.chrome.com/articles/scroll-driven-animations/) for examples.
```
I don't see the point. It is really useful, and removing it from here would just make it harder to find. I know that ideally, it would be nice to have an all-singing, all-dancing tutorial available directly on MDN, but I am kinda loathe to write another one when there is already one available on developer.chrome.com.

chrisdavidmills committed:
Update files/en-us/web/css/@keyframes/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/css_scroll-driven_animations/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/css_scroll-driven_animations/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills commented on files/en-us/web/css/css_scroll-driven_animations/index.md:
```diff
@@ -0,0 +1,84 @@
+
+### Properties
+
+Set the timeline that will control the progress of an animation, and set its attachment range along that timeline:
+
+- {{cssxref("animation-timeline")}}
+- {{cssxref("animation-range")}}
+- {{cssxref("animation-range-start")}}
+- {{cssxref("animation-range-end")}}
+
+Define _named scroll progress timelines_:
+
+- {{cssxref("scroll-timeline")}}
+- {{cssxref("scroll-timeline-axis")}}
+- {{cssxref("scroll-timeline-name")}}
+
+Define _named view progress timelines_:
+
+- {{cssxref("view-timeline")}}
+- {{cssxref("view-timeline-axis")}}
+- {{cssxref("view-timeline-inset")}}
+- {{cssxref("view-timeline-name")}}
```
Consistency is good, but I think breaking long lists up and grouping them with short descriptions is really powerful in situations like this. Otherwise the list becomes really hard to parse and therefore not very useful to the reader. I started to break up lists of Web API interfaces for the same reason, e.g. at https://developer.mozilla.org/en-US/docs/Web/API/WebGPU_API#interfaces.

I think we should do this more rather than less.

chrisdavidmills commented on files/en-us/web/css/css_scroll-driven_animations/index.md:
```diff
@@ -0,0 +1,84 @@
+
+### At-rules
+
+CSS scroll-driven animations adds the ability to include `<timeline-range-name>`s in {{cssxref("@keyframes")}} blocks, to place keyframes at specific positions inside named timeline ranges.
```
The specific reason I added prose here is that without it, it could be read as the Scroll-driven animations spec adding the `@keyframes` at-rule, which isn't true. It only adds a specific sub-feature.

chrisdavidmills committed:
Update files/en-us/web/css/css_scroll-driven_animations/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills commented on files/en-us/web/css/css_scroll-driven_animations/index.md:
```diff
@@ -0,0 +1,84 @@
+- {{domxref("ScrollTimeline")}}
+- {{domxref("ViewTimeline")}}
+
+## Guides
+
+- [Animate elements on scroll with Scroll-driven animations](https://developer.chrome.com/articles/scroll-driven-animations/)
```
Same comment as above; it seems a bit arbitrary to me, and just serves to make this really useful guide harder to find.

dipikabh replied:
For this one, I would still recommend to move the external link to "See also" and remove the "Guides" section. Also, with the recent guidance we added for external links to state when the page was last updated, this would look like:

```
- [Animate elements on scroll with scroll-driven animations](https://developer.chrome.com/articles/scroll-driven-animations/) on developer.chrome.com (May 5, 2023)
```

chrisdavidmills replied:
OK, I've moved the guide to "See also", and removed the `## Guide` heading.

I've updated the link to

> - [Animate elements on scroll with Scroll-driven animations](https://developer.chrome.com/articles/scroll-driven-animations/) on developer.chrome.com

I think listing the site the link is going to is a good idea — it is nice for the user to see that they are about to go to. reputable site, and you could easily automate adding the link text with a couple of lines of script.

However, I REALLY don't think it is a good idea to put last-updated dates on links, because those articles will be updated, and you'll very quickly end up with a maintenance nightmare of out-of-date date references. And we're not really gaining much with these dates either — the user can click through and find out how current the resource is quite easily anyway.

Who was this agreed with, and when? Can I see a discussion thread about it somewhere?

dipikabh replied:
Sure, yeah. This is the discussion thread - https://github.com/orgs/mdn/discussions/339. The thread about dates is here: https://github.com/orgs/mdn/discussions/339#discussioncomment-5173159.
We felt that mentioning dates would be a good idea to let the readers know upfront how old or new the article is.

chrisdavidmills replied:
Thx for the links, I've had a read, and slacked you with a couple of questions.

chrisdavidmills committed:
Merge branch 'main' into scroll-driven-animations

chrisdavidmills commented on files/en-us/web/css/css_scroll-driven_animations/index.md:
```diff
@@ -0,0 +1,84 @@
+
+There are two different types of scroll-based timeline:
+
+- A _scroll progress timeline_ is progressed through by scrolling a scrollable element (_scroller_) between top and bottom (or left and right). The position in the scroll range is converted into a percentage of progress — 0% at the start and 100% at the end.
+- A _view progress timeline_ is progressed through based on the change in visibility of an element (known as the _subject_) inside a scroller. The visibility of the subject inside the scroller is tracked — by default, the timeline is at 0% when the subject is first visible at one edge of the scroller, and 100% when it reaches the opposite edge.
```
Some of your suggestions were good here, but I've updated it in a slightly different way. Specifically:

* I've not included "top and bottom (or left and right) positions in the element", because it sounds like we are talking about something inside the element. We are purely talking about the position of the element's scrollbar.
* I've kept "by default"/ I think this is important because you can change this behavior, for example using `view-timeline-inset`.

dipikabh replied:
Got it 👍

chrisdavidmills committed:
Few tweaks based on dipikabh comments

chrisdavidmills committed:
Merge branch 'main' into scroll-driven-animations

chrisdavidmills committed:
Update guide link position

dipikabh reviewed (changes requested):
Thanks for your updates and responses to the previous review, Chris! I guess only one [comment](https://github.com/mdn/content/pull/27075#discussion_r1229728840) is outstanding.

Going through more pages - mostly nits here. Should hopefully finish going through the remaining docs soon (today).

dipikabh commented on files/en-us/web/css/view-timeline/index.md:
```diff
@@ -0,0 +1,190 @@
+
+The **`view-timeline`** [CSS](/en-US/docs/Web/CSS) [shorthand property](/en-US/docs/Web/CSS/Shorthand_properties) is used to define a _named view progress timeline_, which is progressed through based on the change in visibility of an element (known as the _subject_) inside a scrollable element (_scroller_). `view-timeline` is set on the subject.
+
+The visibility of the subject inside the scroller is tracked — by default, the timeline is at 0% when the subject is first visible at one edge of the scroller, and 100% when it reaches the opposite edge.
+
+`view-timeline` can contain two constituent values — a name for the named view progress timeline, and an optional scroll axis value.
+
+The name is then referenced in an [`animation-timeline`](/en-US/docs/Web/CSS/animation-timeline) declaration to indicate the element that will be animated as the timeline progresses. This can be the subject element, but it doesn't have to be — you can animate a different element as the subject moves through the scrolling area.
```
Removed commas before 'and' and removed line spacing before 'The name is then...' 

```suggestion
The visibility of the subject inside the scroller is tracked — by default, the timeline is at 0% when the subject is first visible at one edge of the scroller and 100% when it reaches the opposite edge.

`view-timeline` can contain two constituent values — a name for the named view progress timeline and an optional scroll axis value.
The name is then referenced in an [`animation-timeline`](/en-US/docs/Web/CSS/animation-timeline) declaration to indicate the element that will be animated as the timeline progresses. This can be the subject element, but it doesn't have to be — you can animate a different element as the subject moves through the scrolling area.
```

dipikabh commented on files/en-us/web/css/view-timeline/index.md:
```diff
@@ -0,0 +1,190 @@
+### Creating a named view progress timeline
+
+A view progress timeline named `--subjectReveal` is defined using the `view-timeline` property on a subject element with a `class` of `animation`.
+This is then set as the timeline for the same element using `animation-timeline: --subjectReveal;`. The result is that the subject element animates as it moves upwards through the document as it is scrolled.
```
```suggestion
This is then set as the timeline for the same element using `animation-timeline: --subjectReveal`. The result is that the subject element animates as it moves upwards through the document as it is scrolled.
```

dipikabh commented on files/en-us/web/css/view-timeline-name/index.md:
```diff
@@ -0,0 +1,171 @@
+
+The **`view-timeline-name`** [CSS](/en-US/docs/Web/CSS) property is used to define the name of a _named view progress timeline_, which is progressed through based on the change in visibility of an element (known as the _subject_) inside a scrollable element (_scroller_). `view-timeline` is set on the subject.
+
+The visibility of the subject inside the scroller is tracked — by default, the timeline is at 0% when the subject is first visible at one edge of the scroller, and 100% when it reaches the opposite edge.
+
+The name is then referenced in an [`animation-timeline`](/en-US/docs/Web/CSS/animation-timeline) declaration to indicate the element that will be animated as the timeline progresses. This can be the subject element, but it doesn't have to be — you can animate a different element as the subject moves through the scrolling area.
```
```suggestion
The visibility of the subject inside the scroller is tracked — by default, the timeline is at 0% when the subject is first visible at one edge of the scroller and 100% when it reaches the opposite edge.
The name is then referenced in an [`animation-timeline`](/en-US/docs/Web/CSS/animation-timeline) declaration to indicate the element that will be animated as the timeline progresses. This can be the subject element, but it doesn't have to be — you can animate a different element as the subject moves through the scrolling area.
```

dipikabh commented on files/en-us/web/css/view-timeline-inset/index.md:
```diff
@@ -0,0 +1,178 @@
+The **`view-timeline-inset`** [CSS](/en-US/docs/Web/CSS) property is used to specify one or two values representing an adjustment to the position of the scrollport (see {{glossary("Scroll container")}} for more details) in which the subject element of a _named view progress timeline_ animation is deemed to be visible. Put another way, this allows you to specify start and/or end inset (or outset)  ...
+
+This can be combined with or used instead of {{cssxref("animation-range")}} and its longhand properties, which can be used to set the attachment range of an animation along its timeline.
+
```
```suggestion
```

dipikabh reviewed (changes requested):
Just some more nits here. I've done a quick readthrough of the CSS pages. Thanks for this huge effort of documenting the scroll-driven spec! 🙌

chrisdavidmills committed:
Update files/en-us/web/css/view-timeline/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/view-timeline/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/view-timeline-name/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/view-timeline-inset/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/animation-range-end/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/animation-range/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/animation-range/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/animation-range/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/animation-timeline/scroll/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/animation-timeline/view/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/animation-timeline/view/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/animation-timeline/view/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/animation-timeline/scroll/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/view-timeline/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/animation-timeline/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/animation-timeline/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/animation-timeline/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/animation-timeline/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Update files/en-us/web/css/animation-timeline/scroll/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

chrisdavidmills committed:
Merge branch 'main' into scroll-driven-animations

chrisdavidmills committed:
Make animation-range* see also sections consistent

chrisdavidmills commented on files/en-us/web/css/animation-range/index.md:
```diff
@@ -0,0 +1,231 @@
+- [`animation-range-start`](/en-US/docs/Web/CSS/animation-range-start) and [`animation-range-end`](/en-US/docs/Web/CSS/animation-range-end) properties
+- [`scroll-timeline`](/en-US/docs/Web/CSS/scroll-timeline), [`scroll-timeline-axis`](/en-US/docs/Web/CSS/scroll-timeline-axis), and [`scroll-timeline-name`](/en-US/docs/Web/CSS/scroll-timeline-name) properties
+- [`animation-timeline`](/en-US/docs/Web/CSS/animation-timeline) property
+- [`view-timeline-inset`](/en-US/docs/Web/CSS/view-timeline-inset) property
```
Done, in my last commit

chrisdavidmills committed:
prettier fix

bramus commented:
Please note that a late change to the spec changed the syntax for `scroll-timeline-name` and `view-timeline-name`. They no longer accept `<custom-ident>` but need a `<dashed-ident>` instead (https://github.com/w3c/csswg-drafts/issues/8746). All code example will also need updating because of this.

Furthermore `timeline-scope` doesn’t seem to be covered. I replaces `scroll-timeline-attachment` and `view-timeline-attachment` which were part of the spec before, but now omitted. (https://github.com/w3c/csswg-drafts/issues/7759)

bramus commented:
And oh, `getCurrentTime` also got removed from scroll-animations-1, as per CSSWG resolution in https://github.com/w3c/csswg-drafts/issues/8765#issuecomment-1581632010. The edit doesn’t seem to be published yet, though.

bramus commented on files/en-us/web/css/scroll-timeline/index.md:
```diff
@@ -103,7 +103,7 @@ The CSS below defines a square that rotates according to the timeline provided b
   height: 100px;
   animation-name: rotateAnimation;
-  animation-duration: 1ms;
+  animation-duration: 1ms; /* Firefox requires this to apply the animation */
```
Same remark as before abou this.

chrisdavidmills replied:
Same response from me too

bramus commented on files/en-us/web/css/view-timeline-axis/index.md:
```diff
@@ -0,0 +1,186 @@
+/* Logical property values */
+view-timeline-axis: block;
+view-timeline-axis: inline;
+/* Older, deprecated non-logical property values */
```
Same remark as for scroll-timeline-axis

chrisdavidmills replied:
Same reply as before ;-)

bramus commented on files/en-us/web/css/view-timeline-name/index.md:
````diff
@@ -0,0 +1,170 @@
+
+```css
+view-timeline-name: none;
+view-timeline-name: custom_name_for_timeline;
````
The names should use `<dashed-ident>`

chrisdavidmills replied:
Note added to make this clear, as with `scroll-timeline-name`

bramus commented on files/en-us/web/css/scroll-timeline-name/index.md:
```diff
@@ -32,8 +29,10 @@ Allowed values for `scroll-timeline-name` are:
 - `none`
   - : The timeline has no name.
 - `<custom-ident>`
```
```suggestion
- `<dashed-ident>`
```

chrisdavidmills replied:
done

bramus commented on files/en-us/web/css/scroll-timeline-name/index.md:
```diff
@@ -32,8 +29,10 @@ Allowed values for `scroll-timeline-name` are:
-  - : An arbitrary custom identifier that can be used to reference the scroll timeline in an [`animation-timeline`](/en-US/docs/Web/CSS/animation-timeline).
-    Possible values are described in [`<custom-ident>`](/en-US/docs/Web/CSS/custom-ident).
+
+  - : An arbitrary custom identifier defining a name for a scroll progress timeline, which can then be referenced in an [`animation-timeline`](/en-US/docs/Web/CSS/animation-timeline) property. See [`<custom-ident>`](/en-US/docs/Web/CSS/custom-ident) for more details about permitted values.
```
```suggestion
  - : An arbitrary custom identifier defining a name for a scroll progress timeline, which can then be referenced in an [`animation-timeline`](/en-US/docs/Web/CSS/animation-timeline) property. See [`<dashed-ident>`](/en-US/docs/Web/CSS/dashed-ident) for more details about permitted values.
```

chrisdavidmills replied:
done

bramus commented on files/en-us/web/css/view-timeline-name/index.md:
```diff
@@ -0,0 +1,170 @@
+  - : The timeline has no name.
+- `<custom-ident>`
+
+  - : An arbitrary custom identifier defining a name for a view progress timeline, which can then be referenced in an [`animation-timeline`](/en-US/docs/Web/CSS/animation-timeline) property. See [`<custom-ident>`](/en-US/docs/Web/CSS/custom-ident) for more details about permitted values.
```
```suggestion
  - : An arbitrary custom identifier defining a name for a view progress timeline, which can then be referenced in an [`animation-timeline`](/en-US/docs/Web/CSS/animation-timeline) property. See [`<dashed-ident>`](/en-US/docs/Web/CSS/dashed-ident) for more details about permitted values.
```

estelle replied:
line 32?

bramus replied:
Line 32 and line 34.

chrisdavidmills replied:
done

chrisdavidmills committed:
Add timeline-scope, remove getCurrentTime(), update dashed-ident details

chrisdavidmills commented:
OK, in the first commit since @bramus reviewed, I have:

- Added a `timeline-scope` page
- Made sure that `scroll-timeline-attachment` and `view-timeline-attachment` are not included anywhere
- Made sure `<dashed-ident>` details plus explanation is included in the `timeline-scope` and `*-timeline-name` pages
- Made sure all examples use dashed idents for the timeline names.
- Removed `getCurrentTime()`

I'll start working through all the above comments tomorrow.

chrisdavidmills committed:
Merge branch 'main' into scroll-driven-animations

chrisdavidmills committed:
Add experimental status to timeline-scope

chrisdavidmills commented on files/en-us/web/api/element/animate/index.md:
```diff
@@ -33,10 +33,9 @@ animate(keyframes, options)
     - `timeline` {{optional_inline}}
-      - : A property unique to `animate()`: the {{domxref("AnimationTimeline")}} to associate with the animation. Defaults to {{domxref("Document.timeline")}}.
+      - : A property unique to `animate()`: The {{domxref("AnimationTimeline")}} to associate with the animation. Defaults to {{domxref("Document.timeline")}}.
 
```
Ah, I originally left them out because they didn't seem to do anything. I think it was around the time when the implementation was in progress. They do seem to work now, although some of the value effects aren't what I'd quite expect. Do all of the value types as specced work, in your experience? 

My descriptions are in my next commit.

bramus replied:
From my experience they seemed to work well. Only used the string variation though, where I – for example – set `rangeEnd` to `"cover 100%"` instead of an object.

chrisdavidmills replied:
Hrm, I wonder whether I should just list the string values then, at least for now, while we investigate the other options.

chrisdavidmills commented on files/en-us/web/api/animationtimeline/index.md:
```diff
@@ -11,9 +11,14 @@ The `AnimationTimeline` interface of the [Web Animations API](/en-US/docs/Web/AP
+- {{domxref("AnimationTimeline.currentTime", "currentTime")}} {{ReadOnlyInline}}
   - : Returns the time value in milliseconds for this timeline or `null` if this timeline is inactive.
 
+## Instance methods
```
Hrm, I deleted this in my last commit, but it seems to be back again today. I've deleted it in my next commit.

chrisdavidmills commented on files/en-us/web/api/scrolltimeline/axis/index.md:
```diff
@@ -0,0 +1,46 @@
+  - : The scrollbar on the block axis of the scroll container, which is the axis in the direction perpendicular to the flow of text within a line. For horizontal writing modes, such as standard English, this is the same as `vertical`, while for vertical writing modes, it is the same as `horizontal`.
+- `inline`
+  - : The scrollbar on the inline axis of the scroll container, which is the axis in the direction parallel to the flow of text in a line. For horizontal writing modes, this is the same as the `horizontal` axis, while for vertical writing modes, this is the same as `vertical`.
+- `vertical`
```
Darnit. I fixed this on the CSS `*-timeline-axis` pages, but forgot about this one. Fixed now.

chrisdavidmills commented on files/en-us/web/api/scrolltimeline/axis/index.md:
```diff
@@ -0,0 +1,46 @@
+  - : The scrollbar on the inline axis of the scroll container, which is the axis in the direction parallel to the flow of text in a line. For horizontal writing modes, this is the same as the `horizontal` axis, while for vertical writing modes, this is the same as `vertical`.
+- `vertical`
+  - : The scrollbar on the vertical axis of the scroll container.
+- `horizontal`
```
Darnit. I fixed this on the CSS `*-timeline-axis` pages, but forgot about this one. Fixed now.

chrisdavidmills commented on files/en-us/web/api/scrolltimeline/scrolltimeline/index.md:
```diff
@@ -0,0 +1,61 @@
+
+        - `block`: The scrollbar on the block axis of the scroll container, which is the axis in the direction perpendicular to the flow of text within a line. For horizontal writing modes, such as standard English, this is the same as `vertical`, while for vertical writing modes, it is the same as `horizontal`.
+        - `inline`: The scrollbar on the inline axis of the scroll container, which is the axis in the direction parallel to the flow of text in a line. For horizontal writing modes, this is the same as the `horizontal` axis, while for vertical writing modes, this is the same as `vertical`.
+        - `vertical`: The scrollbar on the vertical axis of the scroll container.
```
Darnit. I fixed this on the CSS `*-timeline-axis` pages, but forgot about this one. Fixed now.

chrisdavidmills commented on files/en-us/web/api/scrolltimeline/scrolltimeline/index.md:
```diff
@@ -0,0 +1,61 @@
+        - `block`: The scrollbar on the block axis of the scroll container, which is the axis in the direction perpendicular to the flow of text within a line. For horizontal writing modes, such as standard English, this is the same as `vertical`, while for vertical writing modes, it is the same as `horizontal`.
+        - `inline`: The scrollbar on the inline axis of the scroll container, which is the axis in the direction parallel to the flow of text in a line. For horizontal writing modes, this is the same as the `horizontal` axis, while for vertical writing modes, this is the same as `vertical`.
+        - `vertical`: The scrollbar on the vertical axis of the scroll container.
+        - `horizontal`: The scrollbar on the horizontal axis of the scroll container.
```
Darnit. I fixed this on the CSS `*-timeline-axis` pages, but forgot about this one. Fixed now.

chrisdavidmills commented on files/en-us/web/api/viewtimeline/viewtimeline/index.md:
```diff
@@ -0,0 +1,73 @@
+
+        - `"block"`: The scrollbar on the block axis of the scroll container, which is the axis in the direction perpendicular to the flow of text within a line. For horizontal writing modes, such as standard English, this is the same as `vertical`, while for vertical writing modes, it is the same as `horizontal`.
+        - `"inline"`: The scrollbar on the inline axis of the scroll container, which is the axis in the direction parallel to the flow of text in a line. For horizontal writing modes, this is the same as the `horizontal` axis, while for vertical writing modes, this is the same as `vertical`.
+        - `"vertical"`: The scrollbar on the vertical axis of the scroll container.
```
Darnit. I fixed this on the CSS `*-timeline-axis` pages, but forgot about this one. Fixed now.

chrisdavidmills commented on files/en-us/web/api/viewtimeline/viewtimeline/index.md:
```diff
@@ -0,0 +1,73 @@
+        - `"block"`: The scrollbar on the block axis of the scroll container, which is the axis in the direction perpendicular to the flow of text within a line. For horizontal writing modes, such as standard English, this is the same as `vertical`, while for vertical writing modes, it is the same as `horizontal`.
+        - `"inline"`: The scrollbar on the inline axis of the scroll container, which is the axis in the direction parallel to the flow of text in a line. For horizontal writing modes, this is the same as the `horizontal` axis, while for vertical writing modes, this is the same as `vertical`.
+        - `"vertical"`: The scrollbar on the vertical axis of the scroll container.
+        - `"horizontal"`: The scrollbar on the horizontal axis of the scroll container.
```
Darnit. I fixed this on the CSS `*-timeline-axis` pages, but forgot about this one. Fixed now.

chrisdavidmills commented on files/en-us/web/css/animation-timeline/scroll/index.md:
```diff
@@ -9,70 +9,72 @@ browser-compat: css.properties.animation-timeline.scroll
+/* Values for selecting the axis */
+animation-timeline: scroll(block); /* Default */
 animation-timeline: scroll(inline);
 animation-timeline: scroll(vertical);
```
Darnit. I fixed this on the CSS `*-timeline-axis` pages, but forgot about this one. Fixed now.

chrisdavidmills commented on files/en-us/web/css/animation-timeline/scroll/index.md:
```diff
@@ -9,70 +9,72 @@ browser-compat: css.properties.animation-timeline.scroll
+      - : The scrollbar on the block axis of the scroll container, which is the axis in the direction perpendicular to the flow of text within a line. For horizontal writing modes, such as standard English, this is the same as `vertical`, while for vertical writing modes, it is the same as `horizontal`. This is the default value.
+    - `inline`
+      - : The scrollbar on the inline axis of the scroll container, which is the axis in the direction parallel to the flow of text in a line. For horizontal writing modes, this is the same as the `horizontal` axis, while for vertical writing modes, this is the same as `vertical`.
+    - `vertical`
```
Darnit. I fixed this on the CSS `*-timeline-axis` pages, but forgot about this one. Fixed now.

chrisdavidmills commented on files/en-us/web/css/animation-timeline/scroll/index.md:
```diff
@@ -9,70 +9,72 @@ browser-compat: css.properties.animation-timeline.scroll
+      - : The scrollbar on the inline axis of the scroll container, which is the axis in the direction parallel to the flow of text in a line. For horizontal writing modes, this is the same as the `horizontal` axis, while for vertical writing modes, this is the same as `vertical`.
+    - `vertical`
+      - : The scrollbar on the vertical axis of the scroll container.
+    - `horizontal`
```
Darnit. I fixed this on the CSS `*-timeline-axis` pages, but forgot about this one. Fixed now.

chrisdavidmills commented on files/en-us/web/css/animation-timeline/scroll/index.md:
```diff
@@ -103,7 +105,7 @@ Note that the `block nearest` options below are actually the default, so we coul
 
   animation-name: rotateAnimation;
-  animation-duration: 3s;
+  animation-duration: 1ms; /* Firefox requires this to apply the animation */
```
We already discussed this one over email; I think it is worth leaving in for now, and I can remove it later.

chrisdavidmills commented on files/en-us/web/css/animation-timeline/view/index.md:
```diff
@@ -0,0 +1,197 @@
+
+> **Note:** Each use of `view()` corresponds to its own unique instance of {{domxref("ViewTimeline")}} in the [Web Animations API](/en-US/docs/Web/API/Web_Animations_API).
+
+## Syntax
```
Darnit. I fixed this on the CSS `*-timeline-axis` pages, but forgot about this one. Fixed now.

chrisdavidmills commented on files/en-us/web/css/animation-timeline/view/index.md:
```diff
@@ -0,0 +1,197 @@
+animation-timeline: view(100px 200px);
+animation-timeline: view(auto 200px);
+
+/* Examples that specify axis and inset */
```
The function parameters are correct, but the function name still said `scroll` rather than `view`. I've fixed them now.

chrisdavidmills commented on files/en-us/web/css/animation-timeline/view/index.md:
```diff
@@ -0,0 +1,197 @@
+
+### Parameters
+
+- axis
```
Fixed

chrisdavidmills commented on files/en-us/web/css/css_scroll-driven_animations/index.md:
```diff
@@ -0,0 +1,81 @@
+
+{{CSSRef}}
+
+The **CSS scroll-driven animations** module provides functionality that builds on top of the [CSS animations module](/en-US/docs/Web/CSS/CSS_animations) and [Web Animations API](/en-US/docs/Web/API/Web_Animations_API). It allows you to animate property values based on a progression along a scroll-based timeline instead of the default time-based document timeline. This means that you can animate a ...
```
I see what you mean. Updated.

chrisdavidmills commented on files/en-us/web/css/scroll-timeline-axis/index.md:
```diff
@@ -2,47 +2,47 @@
+/* Logical property values */
 scroll-timeline-axis: block;
 scroll-timeline-axis: inline;
+/* Older, deprecated non-logical property values */
```
They did in Firefox

bramus replied:
But the implementation is still behind a flag there, no?

chrisdavidmills replied:
Yeah, you're right, sorry, not a stable release. Do you think I should delete them from the syntax sections so they are a bit less confusing, and then only include them in the code examples where they are needed to make Fx work, like I did with the `animation-duration`?

chrisdavidmills replied:
I've decided to remove the information about the deprecated values from the page, as suggested. I'm also going to open a new browser-compat-data PR to add this info to the Fx support data, so it is covered in the compat tables. Much better place for it.

chrisdavidmills commented on files/en-us/web/css/scroll-timeline/index.md:
```diff
@@ -2,18 +2,18 @@
+
+> **Note:** If the scroller does not overflow its container in the axis dimension or if the overflow is hidden or clipped, no timeline will be created.
 
 ## Constituent properties
```
Dammnit, I changed all this last night, but it looks like somehow a bunch of my changes got lost. I must've been tired. Either that or the linter is buggy.

I've added a note to make this clear.

chrisdavidmills commented on files/en-us/web/css/view-timeline/index.md:
````diff
@@ -0,0 +1,189 @@
+## Syntax
+
+```css
+/* two values: one each for view-timeline-name and view-timeline-axis */
````
Done. I've updated all instances of `custom_name_for_timeline` to `--custom_name_for_timeline`.

chrisdavidmills committed:
Loads of updates

chrisdavidmills committed:
Merge branch 'main' into scroll-driven-animations

chrisdavidmills committed:
Attempt to fix prettier error

chrisdavidmills committed:
Remove deprecated axis values and fix a few errors

chrisdavidmills committed:
Show clear examples of rangeEnd and rangeStart string values

chrisdavidmills committed:
Merge branch 'main' into scroll-driven-animations

chrisdavidmills committed:
Merge branch 'main' into scroll-driven-animations
